racunaljka.js

```diff
@@ -0,0 +1,230 @@
+        }
+         else if(isNaN(pise.value)===false) {
+     pise.value+='-'; 
+      } 
+      else if(isNaN(pise.value)===true) {
+     pise.value+='';
+       } 
+         else{pise.value+='+'}   
+}   
+  
+function puta() {
+    var pise = document.getElementById('izvestaj');
+
+            if(pise.value==='') {
+            pise.value='';
+        }
+         else if(isNaN(pise.value)===false) {
+     pise.value+='x'; 
+      } 
+      else if(isNaN(pise.value)===true) {
+     pise.value+='';
+       } 
+         else{pise.value+='x'}   
+}     
+
+  
+
+function podeljeno() {
+    var pise = document.getElementById('izvestaj');
+
+         if(pise.value==='') {
+            pise.value='';
+        }
+        
+       else if(isNaN(pise.value)===false) {
+     pise.value+='/'; 
+     
+       } 
+     
+    else if(isNaN(pise.value)===true) {
+     pise.value+='';
+       
+       } 
+        
+        else{pise.value+='/'}   
+} 
+
+function jednako() {
+     var pise = document.getElementById('izvestaj');
+     var textLength = pise.value.length;
+for(var i=0; i<textLength; i++) {    
+ if(pise.value[i]==='-'){
+     pise.value1 = pise.value.substring(0,pise.value.indexOf('-'));
+     pise.value2 = pise.value.substring(pise.value.indexOf('-')+1); 
+  
+     pise.value = pise.value1 - pise.value2;
+    } 
+    
+  else if(pise.value[i]==='+'){
+     pise.value1 = pise.value.substring(0,pise.value.indexOf('+'));
+     pise.value2 = pise.value.substring(pise.value.indexOf('+')+1);  
+      pise.value = Number(pise.value1) + Number(pise.value2);
+  }   
+    
+    
+   else if(pise.value[i]==='x'){
+      pise.value1 = pise.value.substring(0,pise.value.indexOf('x'));
+      pise.value2 = pise.value.substring(pise.value.indexOf('x')+1); 
+    
+      pise.value = pise.value1 * pise.value2;  
+   }
+   
+    
+    else if(pise.value[i]==='/'){
+      pise.value1 = pise.value.substring(0,pise.value.indexOf('/'));
+      pise.value2 = pise.value.substring(pise.value.indexOf('/')+1); 
+    
+      pise.value = pise.value1 / pise.value2;  
+   }    
+}
+}
+
+
+/*=================================== operacije================================================*/
+/*=================================== ostalo================================================*/
+function brisanje() {
+    var pise = document.getElementById('izvestaj');
+    pise.value='';
+}
+
+function jednoBrisanje() {
+    var pise = document.getElementById('izvestaj');
+    pise.value=pise.value.substring(0, pise.value.length - 1);
+
+}
+
+function coma() {
+    var pise = document.getElementById('izvestaj');
+    var textLength = pise.value.length;
+    pise.value+='.';
+for(var i=0; i<textLength; i++) {    
+
+ if(pise.value[i]==='.'){
+    pise.value=pise.value.substring(0, pise.value.length - 1);  
+}
+ if(pise.value[i]==='-' || pise.value[i]==='+' || pise.value[i]==='x' || pise.value[i]==='/'){
+    pise.value+='.';
+    
+}
+}
+}
+
+function squareRoot() {
```

Comment:
Try to think how to write a more concise code, since you have a lot of redundancy in most of the methods. Here for example, you can write everything in a much simpler way:
```
function squareRoot() {
  var inputField = document.getElementById('editor');
  if (!isNaN(inputField.value)) {
    inputField.value = Math.sqrt(inputField.value);
  }
}
```